src/translate_ligo.coffee

```diff
@@ -598,9 +598,11 @@ walk = (root, ctx)->
       
       if root.fn.type?.main == "struct"
         # this is contract(address) case
-        msg = "address contract to type_cast is not supported yet (we need enum action type for each contract)"
-        perr "CRITICAL WARNING #{msg}"
-        return "(* #{msg} *)"
+        arg_names = ctx.type_decl_hash[root.fn.name].scope.list
```

Comment:
This is not a fix, this is a hack

Reply:
Please add support for `Struct_init` to ast4gen and handle it in `translate_ligo` in separate `when "Struct_init` clause

Reply:
Type decl hash is a separate entity for typing system and may not be present in compiler at all. The same goes for `root.fn.type?.main` - it was a hack for a particular case, but we should strive for the better modularity.